src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/RazorTranslateDiagnosticsService.cs

```diff
@@ -162,23 +163,34 @@ private Diagnostic[] FilterCSharpDiagnostics(Diagnostic[] unmappedDiagnostics, R
     private static bool InCSharpLiteral(
         Diagnostic d,
         SourceText sourceText,
-        RazorSyntaxTree syntaxTree,
-        ILogger logger)
+        RazorSyntaxTree syntaxTree)
     {
         if (d.Range is null)
         {
             return false;
         }
 
-        var owner = syntaxTree.GetOwner(sourceText, d.Range.End, logger);
-        if (owner is null)
+        
+        var owner = syntaxTree.Root.FindNode(d.Range.AsRazorTextSpan(sourceText), getInnermostNodeForTie: true);
```

Comment:
Hey, I added another `FindNode` usage!

Reply:
That said, I can't explain why it doesn't find the `CSharpCodeBlockSyntax` with the same span. This could be a difference in how Roslyn and Razor make trees/how FindToken works.

My lack of understanding also means I don't think I can make a good comment. I just know it works and all the tests are passing